src/radical/pilot/agent/agent_0.py

```diff
@@ -408,6 +422,10 @@ def _start_services(self):
         self._log.info('create services: %s' % cmdline)
         ru.sh_callout_bg(cmdline, stdout='services.out', stderr='services.err')
 
+        while(all_services_not_started)
+            # WHERe to check state of executing
+            service.state == AGENT_EXECUTING
```

Comment:
State upates are published via ZMQ on the `STATE_PUBSUB`  channel.  Have a look at the Raptor Master again [here](https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/raptor/master.py#L55) and [here](https://github.com/radical-cybertools/radical.pilot/blob/devel/src/radical/pilot/raptor/master.py#L219) on how to subscribe to state updates and how to evaluate them (while here the agent waits for service task state updates, the master waits for worker task updates -- same thing :-) )

Reply:
Someone is publishing the cmd like - 
```        if buckets['raptor']:
            self.publish(rpc.STATE_PUBSUB, {'cmd': 'raptor_state_update',
                                            'arg': buckets['raptor']})
```